.github/workflows/deploy_aks.yml

```diff
@@ -0,0 +1,109 @@
+name: Build and deploy to AKS cluster
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+
+concurrency: deploy-${{ github.ref }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      docker-image-tag: ${{ steps.build-image.outputs.tag }}
+      matrix-environments: ${{ toJSON(steps.*.outputs.matrix-environment) }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Setup ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 3.2.2
+        bundler-cache: true
+    - name: Use Node.js 20.9.0
+      uses: actions/setup-node@v4
+      with:
+        node-version: 20.9.0
+        cache: "npm"
+    - name: Install dependencies
+      run: |
+          sudo apt-get update && sudo apt-get -yqq install libpq-dev
+          bundle install --jobs 4 --retry 3
+          bundle exec rails assets:precompile
+      env:
+          DATABASE_URL: postgres://postgres:password@localhost:5432
+          GOVUK_NOTIFY_API_KEY: fake_key
+          SIGN_IN_METHOD: persona
+
+    - name: Intellisense
+      run: bin/bundle exec yard gems
+
+    - name: Build and push docker image
+      id: build-image
+      uses: DFE-Digital/github-actions/build-docker-image@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        context: .
+      env:
+        SIGN_IN_METHOD: persona
+
+    - name: Prepare matrix environments review
+      id: matrix-env-review
+      if: github.event_name == 'pull_request'
+      run: echo "matrix-environment=review" >> $GITHUB_OUTPUT
+
+  deploy:
```

Comment:
we normally have review deployment as it's own job i.e. deploy_review. It should run only if a deploy label is added to the PR. So 'if: contains(github.event.pull_request.labels.*.name, 'deploy')'

Reply:
added condition